### PR TITLE
RC2 argument name changes

### LINKED ISSUE #20501: Breaking Change: Fix argument name in CompareInfo.IsPrefix (suffix -> prefix)
## Fix argument name in CompareInfo.IsPrefix (suffix -> prefix)

This `CompareInfo.IsPrefix()` API was added in .NET 5.0 RC1, but there is a mistake in the argument name where `suffix` was used instead of `prefix`.  The implementation is correct, but the reference assembly was wrong. dotnet/runtime#41803 fixed the argument name. This fix will ship with .NET 5.0 RC2.

### Version introduced

.NET 5.0 RC2

### Old behavior

The previous method signature was the following, with an argument named `suffix`.

```csharp
public bool IsPrefix(System.ReadOnlySpan<char> source, System.ReadOnlySpan<char> suffix, System.Globalization.CompareOptions options, out int matchLength)
```

### New behavior

The updated method signature is the following, with the argument renamed to `prefix`.

```csharp
public bool IsPrefix(System.ReadOnlySpan<char> source, System.ReadOnlySpan<char> prefix, System.Globalization.CompareOptions options, out int matchLength)
```

### Reason for change

This corrects a mistake in the argument name and will improve the API, avoiding confusion by users.

### Recommended action

If the argument was being used as a named argument, update calling code to correct the argument name to match the renamed `prefix` name.

### Category

- [ ] ASP.NET Core
- [ ] C#
- [ ] Code analysis
- [x] Core .NET libraries
- [ ] Cryptography
- [ ] Data
- [ ] Debugger
- [ ] Deployment
- [ ] Globalization
- [ ] Interop
- [ ] JIT
- [ ] LINQ
- [ ] Managed Extensibility Framework (MEF)
- [ ] MSBuild
- [ ] Networking
- [ ] Printing
- [ ] Security
- [ ] Serialization
- [ ] Visual Basic
- [ ] Windows Forms
- [ ] Windows Presentation Foundation (WPF)
- [ ] XML, XSLT

### Affected APIs

[CompareInfo.IsPrefix(ReadOnlySpan&lt;char&gt;, ReadOnlySpan&lt;char&gt;, CompareOptions, out int)](https://docs.microsoft.com/en-us/dotnet/api/system.globalization.compareinfo.isprefix?view=net-5.0)

The affected overload was added in .NET 5.0 RC1 and was not yet present in documentation at the time of this breaking change in RC2.

---
#### Issue metadata

* Issue type: breaking-change

### LINKED ISSUE #20502: Breaking Change: Fix param name in ActivityContext..ctor (traceOptions -> traceFlags)
## Fix param name in ActivityContext..ctor (traceOptions -> traceFlags)

The `ActivityContext` API was added earlier in .NET 5.0 Preview 4, but there was a mistake in the constructor's argument name where `traceOptions` was used instead of `traceFlags`.  The implementation is correct but the reference assembly was wrong. dotnet/runtime#41843 fixed the argument name and this fix will ship with .NET 5.0 RC2.

### Version introduced

.NET 5.0 RC2

### Old behavior

The previous constructor signature was the following, with an argument named `traceOptions`.

```csharp
public ActivityContext(System.Diagnostics.ActivityTraceId traceId, System.Diagnostics.ActivitySpanId spanId, System.Diagnostics.ActivityTraceFlags traceOptions, string? traceState = null)

```

### New behavior

The updated constructor signature is the following, with the argument renamed to `traceFlags`.

```csharp
public ActivityContext(System.Diagnostics.ActivityTraceId traceId, System.Diagnostics.ActivitySpanId spanId, System.Diagnostics.ActivityTraceFlags traceFlags, string? traceState = null, bool isRemote = false)
```

### Reason for change

This corrects a mistake in the argument name and will improve the API, avoiding confusion by users.

### Recommended action

If the argument was being used as a named argument, update calling code to correct the argument name to match the renamed `traceFlags` name.

### Category

- [ ] ASP.NET Core
- [ ] C#
- [ ] Code analysis
- [x] Core .NET libraries
- [ ] Cryptography
- [ ] Data
- [ ] Debugger
- [ ] Deployment
- [ ] Globalization
- [ ] Interop
- [ ] JIT
- [ ] LINQ
- [ ] Managed Extensibility Framework (MEF)
- [ ] MSBuild
- [ ] Networking
- [ ] Printing
- [ ] Security
- [ ] Serialization
- [ ] Visual Basic
- [ ] Windows Forms
- [ ] Windows Presentation Foundation (WPF)
- [ ] XML, XSLT

### Affected APIs

[ActivityContext(ActivityTraceId, ActivitySpanId, ActivityTraceFlags, String, Boolean)](https://docs.microsoft.com/en-us/dotnet/api/system.diagnostics.activitycontext.-ctor?view=net-5.0)

---
#### Issue metadata

* Issue type: breaking-change

### PR DESCRIPTION
**Pull request by gewarren**
Fixes #20501; fixes #20502.

[Preview link](https://review.docs.microsoft.com/en-us/dotnet/core/compatibility/corefx?branch=pr-en-us-20677#parameter-names-changed-in-rc1).

### CODE REVIEW

File: includes/core-changes/corefx/5.0/reference-assembly-parameter-names.md [hunk 1]
```diff
@@ -10,24 +10,26 @@ In .NET 5.0, these mismatched parameter names were updated in the reference asse
...
-| <xref:System.ComponentModel.RefreshPropertiesAttribute.Equals(System.Object)?displayProperty=nameWithType> | `value` | `obj` |
-| <xref:System.Diagnostics.StackFrame.%23ctor(System.Boolean)> | `fNeedFileInfo` | `needFileInfo` |
-| <xref:System.Diagnostics.StackFrame.%23ctor(System.Int32,System.Boolean)> | `fNeedFileInfo` | `needFileInfo` |
-| <xref:System.StringNormalizationExtensions.IsNormalized(System.String,System.Text.NormalizationForm)?displayProperty=nameWithType> | `value` | `strInput` |
-| <xref:System.StringNormalizationExtensions.IsNormalized(System.String)?displayProperty=nameWithType> | `value` | `strInput` |
-| <xref:System.StringNormalizationExtensions.Normalize(System.String,System.Text.NormalizationForm)?displayProperty=nameWithType> | `value` | `strInput` |
-| <xref:System.StringNormalizationExtensions.Normalize(System.String)?displayProperty=nameWithType> | `value` | `strInput` |
+| <xref:System.Diagnostics.ActivityContext.%23ctor(System.Diagnostics.ActivityTraceId,System.Diagnostics.ActivitySpanId,System.Diagnostics.ActivityTraceFlags,System.String,System.Boolean)> | `traceOptions` | `traceFlags` | .NET 5.0 RC2 |
```
Comment (gewarren) [inline includes/core-changes/corefx/5.0/reference-assembly-parameter-names.md:15]: This is a new row in the table.

File: includes/core-changes/corefx/5.0/reference-assembly-parameter-names.md [hunk 2]
```diff
@@ -10,24 +10,26 @@ In .NET 5.0, these mismatched parameter names were updated in the reference asse
...
+| <xref:System.CodeDom.Compiler.CodeGenerator.GenerateStatements(System.CodeDom.CodeStatementCollection)?displayProperty=nameWithType> | `stms` | `stmts` | .NET 5.0 |
+| <xref:System.Globalization.CompareInfo.IsPrefix(System.ReadOnlySpan{System.Char},System.ReadOnlySpan{System.Char},System.Globalization.CompareOptions,System.Int32@)?displayProperty=nameWithType> | `suffix` | `prefix` | .NET 5.0 RC2 |
```
Comment (gewarren) [inline includes/core-changes/corefx/5.0/reference-assembly-parameter-names.md:17]: This is a new row in the table.

**Review by jeffhandley — APPROVED** [review]
Looks good, but I have a question.

The preview link shows these as part of the breaking changes between 3.1 and 5.0.  These were only breaking from an earlier 5.0 release as the APIs were introduced in 5.0.  It looks like there isn't a page for 5.0 RC1 (or earlier) to RC2.  Was it intentional to add them to the 3.1 to 5.0 page since those pages don't exist?

**Comment by gewarren** [comment]
> Looks good, but I have a question.
> 
> The preview link shows these as part of the breaking changes between 3.1 and 5.0. These were only breaking from an earlier 5.0 release as the APIs were introduced in 5.0. It looks like there isn't a page for 5.0 RC1 (or earlier) to RC2. Was it intentional to add them to the 3.1 to 5.0 page since those pages don't exist?

Yes, that's a valid point. I'll create a separate breaking change entry for these two parameters, then. It will live in the [Core .NET libraries breaking changes article](https://review.docs.microsoft.com/en-us/dotnet/core/compatibility/corefx?branch=pr-en-us-20677#parameter-names-changed-in-reference-assemblies) file.

**Review by BillWagner — APPROVED** [review]
This LGTM @gewarren 

You can :shipit: when ready.

**Comment by gewarren** [comment]
Thanks for the reviews!

**Pull request merged**